components/google-cloud/google_cloud_components/mb_sdk_component/remote_runner.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for remote execution of Model Builder SDK as a pipeline component."""
+
+import argparse
+import inspect
+import os
+from google.cloud import aiplatform
+from google.cloud import storage
+
+INIT_KEY = 'init'
+METHOD_KEY = 'method'
+
+
+def split_args(kwargs):
+  """Split args to match MB SDK input format for init, method and config."""
+  init_args = {}
+  method_args = {}
+  config_args = {}
+
+  for key, arg in kwargs.items():
+    if key.startswith(INIT_KEY):
+      init_args[key.split('.')[-1]] = arg
+    elif key.startswith(METHOD_KEY):
+      method_args[key.split('.')[-1]] = arg
+    else:
+      config_args[key.split('.')[-1]] = arg
+
+  return init_args, method_args, config_args
+
+
+def write_to_gcs(project, gcs_uri, text):
+  """Helper method to write resouce names to gcs for downstream components."""
+
+  gcs_uri = gcs_uri[5:]
+  gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
+
+  with open('resource_name.txt', 'w') as f:
+    f.write(text)
+
+  client = storage.Client(project=project)
+  bucket = client.get_bucket(gcs_bucket)
+  blob = bucket.blob(os.path.join(gcs_blob, 'resource_name.txt'))
+  blob.upload_from_filename('resource_name.txt')
+
+
+def read_from_gcs(project, gcs_uri):
+  gcs_uri = gcs_uri[5:]
+  gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
+  client = storage.Client(project=project)
+  bucket = client.get_bucket(gcs_bucket)
+  blob = bucket.get_blob(os.path.join(gcs_blob, 'resource_name.txt'))
+  resource_name = blob.download_as_string().decode('utf-8')
+  return resource_name
+
+
+def resolve_project(serialized_args):
+  return serialized_args['init'].get('project',
```

Comment:
```suggestion
  return serialized_args[INIT_KEY].get('project',
```

Reply:
Done, also applied the same for `method` key.